honeybee_radiance_command/options/gensky.py

```diff
@@ -0,0 +1,256 @@
+# coding: utf-8
+
+from .optionbase import (
+    OptionCollection,
+    BoolOption,
+    NumericOption,
+    TupleOption,
+    ToggleOption
+)
+import warnings
+
+
+class GenskyOptions(OptionCollection):
+    """gensky command options.
+
+    Also see: https://floyd.lbl.gov/radiance/man_html/gensky.1.html
+    """
+
+    __slots__ = (
+        "_s",
+        "_c",
+        "_i",
+        "_u",
+        "_g",
+        "_b",
+        "_B",
+        "_r",
+        "_R",
+        "_t",
+        "_a",
+        "_o",
+        "_m",
+        "_ang")
+
+    def __init__(self):
+        """Gensky command options."""
+
+        OptionCollection.__init__(self)
+
+        # self._s = BoolOption("s", "Sunny sky without sun - default: False")
+        self._s = ToggleOption("s", "Sunny sky", value=None)
+        self._c = BoolOption("c", "Cloudy sky")
+        self._i = ToggleOption("i", "Intermediate sky", value=None)
+        self._u = BoolOption("u", "Uniform cloudy sky")
+        self._g = NumericOption("g", "Average ground reflectance")
+        self._b = NumericOption("b", "Zenith brightness computed from sun and"
+                                " sky turbidity.")
+        self._B = NumericOption("B", "Zenith brightness computed from"
+                                " horizontal diffuse irradiance.")
+        self._r = NumericOption("r", "Solar radiance computed from solar altitude.")
+        self._R = NumericOption("R", "Solar radiance computed from"
+                                " horizontal direct irradiance.")
+        self._t = NumericOption("t", "Turbuity factor", min_value=1.0)
+        self._a = NumericOption("a", "Site latitude", min_value=-90.0, max_value=90.0)
+        self._o = NumericOption("o", "Site longitude", min_value=-180.0,
+                                max_value=180.0)
+        self._m = NumericOption("m", "Standard meridian", min_value=-18.0,
+                                max_value=18.0)
+        self._ang = TupleOption("ang", "Altitude & azimuth", value=None, length=2,
+                                numtype=float)
+        self._on_setattr_check = True
+
+    def _on_setattr(self):
+        """This method executes after setting each new attribute.
+
+        Use this method to add checks that are necessary for OptionCollection. 
+        """
+        if self._ang.is_set and (
+                self._a.is_set or self._o.is_set or self._m.is_set):
+            warnings.warn(
+                'Options -a, -o and -m do not apply when -ang is set.'
+            )
+
```

Comment:
Shouldn't we add another check to ensure the sky is either sunny or intermediate or cloudy?

If I understand correctly `-s -u` should be considered invalid.